cli/cli.go

```diff
@@ -614,8 +622,23 @@ func runPromptActions(p ui.Prompter, o *CliCmdOptions, actions ...promptAction)
 	return nil
 }
 
+func selectNodeSetup(p ui.Prompter, o *CliCmdOptions) (err error) {
+	options := []string{EthereumNode, LidoNode}
+	index, err := p.Select("Select node setup", "", options)
+	if err != nil {
+		return err
+	}
+	o.nodeSetup = options[index]
+	return
+}
+
 func selectNetwork(p ui.Prompter, o *CliCmdOptions) error {
-	options := []string{NetworkMainnet, NetworkSepolia, NetworkGnosis, NetworkChiado, NetworkHolesky}
+	var options []string
+	if o.nodeSetup == LidoNode {
+		options = []string{NetworkMainnet, NetworkHolesky, NetworkSepolia}
+	} else {
+		options = []string{NetworkMainnet, NetworkSepolia, NetworkGnosis, NetworkChiado, NetworkHolesky}
+	}
```

Comment:
Please refactor this to use something like the Factory Method design pattern. This will make it very easy and extensible to add a new `nodeSetup` later on. 

For you to have an overview of the idea, the code would look similar to the following:


```
// NodeOptionsFactory is the interface for creating options
type NodeOptionsFactory interface {
	CreateOptions() []string
}

// GetNodeOptions returns the appropriate NodeOptionsFactory based on node setup
func GetNodeOptions(nodeSetup string) NodeOptionsFactory {
	switch nodeSetup {
	case LidoNode:
		return &LidoNodeOptions{}
	default:
		return &EthereumNodeOptions{}
	}
}

func selectNetwork(p ui.Prompter, o *CliCmdOptions) error {
	o := GetNodeOptions(o.nodeSetup)
	options := o.CreateOptions()
        index, err := p.Select("Select network", "", options)
	if err != nil {
		return err
	}
	o.genData.Network = options[index]
	return nil
}
```

Any other functional approach or design pattern that helps us access the `options` in an extensible way would also be appreciated.

Reply:
[Done](https://github.com/NethermindEth/sedge/pull/389/commits/381a99f8d4edc125d9a267655d844c558d02a21d)